torch_xla/csrc/ops/device_data.h

```diff
@@ -22,6 +22,15 @@ class DeviceData : public XlaNode {
     return data_;
   }
 
+  // With SPMD sharding propagation, we need to update the unpartitioned
+  // backend data with a partitioned one in the node operands.
+  void Assign(std::shared_ptr<torch::lazy::BackendData> data) {
+    XLA_CHECK(data->shape() == data_->shape())
```

Comment:
nit, let's also checkout about `!data_->hasValue()`, we should only call Assign on placeholders.

Reply:
I take this as a short term solution. In the long term we should figure out why DeviceData is being generated by on a placeholder during a `SyncTensorGraphsInternal` call.

Reply:
Ah, for the current use case it's true -- do we want to restrict the API like that? Alternatively, we can pass the ownership (move) if the data is not a placeholder if someone tries to use the API. Hmm I am split, and see that the restriction would be safer semantically. WDYT?

Reply:
Will restrict it so that this is only used for placholders until we find more use cases (justification) cc @JackCaoG